src/core/i18n/utils.js

```diff
@@ -278,19 +278,29 @@ export function makeI18n(
     i18n.options._momentDefineLocale();
   }
 
-  // This makes sure moment always uses the current locale.
-  moment.locale(makeMomentLocale(i18n.lang));
-
-  // Wrap the core Jed functionality so that we can always strip leading whitespace
-  // from translation keys to match the same process used in extraction.
+  // Wrap the core Jed functionality so that we can always strip leading
+  // whitespace from translation keys to match the same process used in
+  // extraction.
   i18n._dcnpgettext = i18n.dcnpgettext;
   i18n.dcnpgettext = function dcnpgettext(domain, context, singularKey, pluralKey, val) {
     return i18n._dcnpgettext(domain, context, oneLineTranslationString(singularKey),
       oneLineTranslationString(pluralKey), val);
   };
 
-  // We add a translated "moment" property to our `i18n` object
-  // to make translated date/time/etc. easy.
-  i18n.moment = moment;
+  const momentLocale = makeMomentLocale(i18n.lang);
+
+  // We add a translated "moment" property to our `i18n` object to make
+  // translated date/time/etc. easy.
+  i18n.moment = (...params) => {
+    const scopedMoment = moment(...params);
+
+    // This also makes sure moment always uses the current locale.
+    scopedMoment.locale(momentLocale);
+
+    return scopedMoment;
+  };
+
+  i18n.moment.locale = () => momentLocale;
```

Comment:
Can't you just update the tests or am I missing something?

```diff
diff --git a/src/core/i18n/utils.js b/src/core/i18n/utils.js
index 5dabcb426..8c714a640 100644
--- a/src/core/i18n/utils.js
+++ b/src/core/i18n/utils.js
@@ -300,7 +300,5 @@ export function makeI18n(
     return scopedMoment;
   };
 
-  i18n.moment.locale = () => momentLocale;
-
   return i18n;
 }
diff --git a/tests/unit/core/i18n/test_utils.js b/tests/unit/core/i18n/test_utils.js
index 156e83bb4..a92a3bf3c 100644
--- a/tests/unit/core/i18n/test_utils.js
+++ b/tests/unit/core/i18n/test_utils.js
@@ -418,7 +418,7 @@ describe('i18n utils', () => {
         },
       };
       const i18n = utils.makeI18n(i18nData, 'fr', FakeJed);
-      expect(i18n.moment.locale()).toEqual('fr');
+      expect(i18n.moment().locale()).toEqual('fr');
       sinon.assert.called(i18nData.options._momentDefineLocale);
     });
 
@@ -431,7 +431,7 @@ describe('i18n utils', () => {
       };
 
       const i18n = utils.makeI18n(i18nData, 'en', FakeJed);
-      expect(i18n.moment.locale()).toEqual('en');
+      expect(i18n.moment().locale()).toEqual('en');
     });
 
     it('always passes the locale to moment', () => {
@@ -442,7 +442,7 @@ describe('i18n utils', () => {
         },
       };
       const i18n = utils.makeI18n(i18nData, 'fr', FakeJed);
-      expect(i18n.moment.locale()).toEqual('fr');
+      expect(i18n.moment().locale()).toEqual('fr');
     });
 
     it('formats a number', () => {
@@ -484,7 +484,7 @@ describe('i18n utils', () => {
       const i18n = utils.makeI18n({}, 'fr', FakeJed);
       // We modify the moment locale, globally.
       moment.locale('de');
-      expect(i18n.moment.locale()).toEqual('fr');
+      expect(i18n.moment().locale()).toEqual('fr');
     });
   });
 });
```

Reply:
Ah well, if that works, I'd be very happy to use this. I must admit, I did not find anything in the doc and went for the easy solution, forgetting to try this before 😞

Reply:
Bwaaa I am stuuuupid. This is perfectly fine, thanks for this, this is much better!